tests/js/destructuring/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -79,12 +77,7 @@ const UserComponent = function ({
   return;
 };
 
-const {
-  a,
-  b,
-  c,
-  d: { e },
-} = someObject;
+const { a, b, c, d: { e } } = someObject;
```

Comment:
This is side effect of https://github.com/prettier/prettier/pull/9288/files#diff-3abca41914b4b214a2507e33cc54c28bR1096, I think it's okay to change, or even preferred.

**Prettier 2.1.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEaIFs4DyARgFZxhYC8mwmAhgDSZFNhMAmSNmcmAvv0wBuADpRIUDN0bNWHLrV4CB1HPmJkKQkAxAQADjACW6ZKDoAnCxADuABUsI0yEHQA2NugE9nuohbowAGs4GABlfUCjKABzZBgLAFc4XThcIjh2dkyAGTpYxLoYuAAxCAtcOhhjWJc6RJgIHRAACxhcNwB1FqN4NEiwODCnXqMAN16vFzA0XxBotDgLGDsAmMrkADN3Rd0SNAAPACEA4NCwunwc6LgtnZSQfYOw6Ji3OABFRIh4O7ddkCRCyLCwuIh0DJuZr6CzRGCdIzsGAtZAADgADLoYRBFp0AvoXDC4CCxrddABHb7wVYGZwoOhoAC0UDgmUyzQscEpRk5qyKGyQ23+D0WuCM8SSIte7y+P1ugvuuhgEIRSJRSAATEqAkY3K8AMJ4AUgYkAVmaiUWABUIXShQCxskAJJQbKwMJgWGGACCrrCMC87z+iz4fCAA)
```sh
--parser babel
```

**Input:**
```jsx
const someObject = { a, b, c, d: { e } } ;
const { a, b, c, d: { e } } = someObject;
```

**Output:**
```jsx
const someObject = { a, b, c, d: { e } };
const {
  a,
  b,
  c,
  d: { e },
} = someObject;

```

**Prettier pr-9288**
[Playground link](https://deploy-preview-9288--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuc0DOMAEaIFs4DyARgFZxhYC8mwmAhgDSZFNhMAmSNmcmAvv0wBuADpRIUDN0bNWHLrV4CB1HPmJkKQkAxAQADjACW6ZKDoAnCxADuABUsI0yEHQA2NugE9nuohbowAGs4GABlfUCjKABzZBgLAFc4XThcIjh2dkyAGTpYxLoYuAAxCAtcOhhjWJc6RJgIHRAACxhcNwB1FqN4NEiwODCnXqMAN16vFzA0XxBotDgLGDsAmMrkADN3Rd0SNAAPACEA4NCwunwc6LgtnZSQfYOw6Ji3OABFRIh4O7ddkCRCyLCwuIh0DJuZr6CzRGCdIzsGAtZAADgADLoYRBFp0AvoXDC4CCxrddABHb7wVYGZwoOhoAC0UDgmUyzQscEpRk5qyKGyQ23+D0WuCM8SSIte7y+P1ugvuuhgEIRSJRSAATEqAkY3K8AMJ4AUgYkAVmaiUWABUIXShQCxskAJJQbKwMJgWGGACCrrCMC87z+iz4fCAA)
```sh
--parser babel
```

**Input:**
```jsx
const someObject = { a, b, c, d: { e } } ;
const { a, b, c, d: { e } } = someObject;
```

**Output:**
```jsx
const someObject = { a, b, c, d: { e } };
const { a, b, c, d: { e } } = someObject;

```

Reply:
I'm sure this was discussed a lot, probably in #2550.

Reply:
Thanks for the linked issue, I realized this is format like that on purpose.

It makes this PR  meaningless, assign in `for-of` is much less use than declaration.

The reason I  pick this issue so quickly is during https://github.com/prettier/prettier/pull/9225, I found similar problem too, I thought it's bug.